src/server/service/kv.rs

```diff
@@ -341,6 +437,247 @@ impl<E: Engine, L: LockMgr> Batcher<E, L> for ReadBatcher {
     }
 }
 
+#[derive(Hash, PartialEq, Eq, Debug)]
+struct WriteId {
+    region_id: RegionVerId,
+    sync_log: bool,
+}
+
+impl WriteId {
+    fn from_context(ctx: &Context) -> Self {
+        WriteId {
+            region_id: RegionVerId::from_context(ctx),
+            sync_log: ctx.get_sync_log(),
+        }
+    }
+}
+
+struct WriteBatchMeta {
+    idx: usize,
+    key_set: HashSet<Vec<u8>>,
+    // TODO(tabokie): size limit here
+}
+
+impl WriteBatchMeta {
+    fn new() -> Self {
+        WriteBatchMeta {
+            idx: 0,
+            key_set: HashSet::default(),
+        }
+    }
+}
+
+struct WriteBatcher {
+    cmd: BatchableRequestKind,
+    commands: Vec<Command>,
+    router: HashMap<WriteId, WriteBatchMeta>,
```

Comment:
Why not we map WriteId to (WriteBatchMeta, Command) directly?

Reply:
make sense